pkg/cache/cache.go

```diff
@@ -163,6 +163,11 @@ func (c *Cache) PutNar(_ context.Context, hash, compression string, r io.ReadClo
 	return err
 }
 
+// DeleteNar deletes the nar from the store.
+func (c *Cache) DeleteNar(_ context.Context, hash, compression string) error {
```

Comment:
How is that authenticated?

Reply:
It's not! I am however adding a flag to the server (in this pr, working on it) `--allow-deletes` for MVP version. We can add basic auth or some other auth later. I'm not planning to use it, but it's here if needed.